From e63c57b65a86dbb5734b9e098945a6c771641ae2 Mon Sep 17 00:00:00 2001
From: Yuchou Gan <yuchou.gan@nxp.com>
Date: Wed, 21 Dec 2016 02:09:19 +0800
Subject: [PATCH 1367/5242] MGS-2483 Increase the gpu 2d and 3d frequency to
 475M Hz on 7ULP

commit  786f959f88ed008f9a3d2b8b5d8f87f530ce9dcb from
https://source.codeaurora.org/external/imx/linux-imx.git

Increase the gpu 2d and 3d frequency to 475M Hz on 7ULP

Signed-off-by: Yuchou Gan <yuchou.gan@nxp.com>
Date: Dec 20, 2016
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/clk/imx/clk-imx7ulp.c |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/clk/imx/clk-imx7ulp.c b/drivers/clk/imx/clk-imx7ulp.c
index 225d1c1..06bdd84 100644
--- a/drivers/clk/imx/clk-imx7ulp.c
+++ b/drivers/clk/imx/clk-imx7ulp.c
@@ -175,6 +175,10 @@ static void __init imx7ulp_clocks_init(struct device_node *scg_node)
 
 	for (i = 0; i < ARRAY_SIZE(clks_init_on); i++)
 		imx_clk_prepare_enable(clks[clks_init_on[i]]);
+        imx_clk_set_parent(clks[IMX7ULP_CLK_GPU2D], clks[IMX7ULP_CLK_APLL_PFD2]);
+        imx_clk_set_parent(clks[IMX7ULP_CLK_GPU3D], clks[IMX7ULP_CLK_APLL_PFD2]);
+        imx_clk_prepare_enable(clks[IMX7ULP_CLK_GPU3D]);
+        imx_clk_prepare_enable(clks[IMX7ULP_CLK_GPU2D]);
 
 	pr_info("i.MX7ULP clock tree init done.\n");
 }
-- 
1.7.9.5

